### PR TITLE
Add support fo java.time, update deps

### PR DESCRIPTION
**Pull request by ikitommi**

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -13,15 +13,15 @@
...
           :output-path "doc"
           :metadata {:doc/format :markdown}}
-  :dependencies [[com.fasterxml.jackson.core/jackson-databind "2.9.4"]]
+  :dependencies [[com.fasterxml.jackson.core/jackson-databind "2.9.5"]
+                 [com.fasterxml.jackson.datatype/jackson-datatype-jsr310 "2.9.5"]]
   :profiles {:dev {:dependencies [[org.clojure/clojure "1.9.0"]
+                                  [com.fasterxml.jackson.datatype/jackson-datatype-joda "2.9.5"]
                                   [cheshire "5.8.0"]
-                                  [criterium "0.4.4"]]
-                   :global-vars {*warn-on-reflection* true}}
```
Comment (miikka) [inline project.clj]: `*warn-on-reflection*` was on for a purpose: for performance, we don't want to do any calls via reflection.

File: src/clj/jsonista/core.clj [hunk 2]
```diff
@@ -122,9 +127,16 @@
...
      (cond-> (:pretty options) (.enable SerializationFeature/INDENT_OUTPUT)
-             (:escape-non-ascii options) (.enable ^"[Lcom.fasterxml.jackson.core.JsonGenerator$Feature;"
-                                                  (into-array [JsonGenerator$Feature/ESCAPE_NON_ASCII]))))))
+             (:escape-non-ascii options) (doto (-> .getFactory (.enable JsonGenerator$Feature/ESCAPE_NON_ASCII))))
+     (as-> $
+           (if-let [modules (:modules options)]
+             (loop [[module rest] modules, mapper $]
+               (let [mapper (.registerModule mapper module)]
+                 (if (seq rest) (recur rest mapper) mapper)))
+             $))
```
Comment (miikka) [inline src/clj/jsonista/core.clj]: Why not just:

```clojure
(doseq [module (:modules options)]
  (.registerModule $ module))
```

**Review by miikka — APPROVED** [review]
Supporting java.time and documenting how to support Joda-Time is good. Added some comments, though.

File: project.clj (hunk 1 above)
Comment (ikitommi) [reply project.clj]: Problem is that with Cursive, there are a lot of false(?) positives with that setting:

```
Reflection warning, /private/var/folders/39/j64jmzc96xdd40ghbvw72rk80000gn/T/form-init6952961206932838438.clj:1:180 - reference to field getFileName can't be resolved.
Reflection warning, /private/var/folders/39/j64jmzc96xdd40ghbvw72rk80000gn/T/form-init6952961206932838438.clj:1:219 - reference to field getLineNumber can't be resolved.
Reflection warning, /private/var/folders/39/j64jmzc96xdd40ghbvw72rk80000gn/T/form-init6952961206932838438.clj:1:261 - reference to field getClassName can't be resolved.
Reflection warning, /private/var/folders/39/j64jmzc96xdd40ghbvw72rk80000gn/T/form-init6952961206932838438.clj:1:431 - reference to field getStackTrace can't be resolved.
Reflection warning, /private/var/folders/39/j64jmzc96xdd40ghbvw72rk80000gn/T/form-init6952961206932838438.clj:1:705 - reference to field getClassName can't be resolved.
Reflection warning, /private/var/folders/39/j64jmzc96xdd40ghbvw72rk80000gn/T/form-init6952961206932838438.clj:1:692 - call to method startsWith can't be resolved (target class is unknown).
Reflection warning, /private/var/folders/39/j64jmzc96xdd40ghbvw72rk80000gn/T/form-init6952961206932838438.clj:1:765 - reference to field getStackTrace can't be resolved.
Reflection warning, /private/var/folders/39/j64jmzc96xdd40ghbvw72rk80000gn/T/form-init6952961206932838438.clj:1:2287 - reference to field getName can't be resolved.
Reflection warning, /private/var/folders/39/j64jmzc96xdd40ghbvw72rk80000gn/T/form-init6952961206932838438.clj:1:4562 - reference to field flush can't be resolved.
```

File: src/clj/jsonista/core.clj (hunk 2 above)
Comment (ikitommi) [reply src/clj/jsonista/core.clj]: fixed, thanks!

File: project.clj (hunk 1 above)
Comment (miikka) [reply project.clj]: That's weird.

**Pull request merged**